apps/admin-ui/src/spa/ReservationUnit/edit/index.tsx

```diff
@@ -1071,60 +1084,104 @@ function ReservationUnitSettings({
           style={{ gridColumn: "1 / -1" }}
         >
           <Grid>
-            <Span6>
-              <ActivationGroup
-                label={t("ReservationUnitEditor.bufferTimeBefore")}
+            <Span12>
+              <Controller
                 control={control}
-                name="hasBufferTimeBefore"
-              >
-                <Controller
-                  control={control}
-                  name="bufferTimeBefore"
-                  render={({ field: { value, onChange } }) => (
-                    <Select
-                      id="bufferTimeBefore"
-                      options={bufferTimeOptions}
-                      label={t(
-                        "ReservationUnitEditor.bufferTimeBeforeDuration"
+                name="reservationBlockWholeDay"
+                render={({ field: { value, onChange } }) => (
+                  <SelectionGroup
+                    errorText={getTranslatedError(
+                      t,
+                      errors.reservationBlockWholeDay?.message
+                    )}
+                    defaultValue="no-buffer"
+                  >
+                    <RadioButton
+                      id="no-buffer"
+                      value="no-buffer"
+                      label={t("ReservationUnitEditor.noBuffer")}
+                      onChange={(e) => onChange(e.target.value)}
+                      checked={value != null && value === "no-buffer"}
+                    />
+                    <RadioButton
+                      id="blocks-whole-day"
+                      value="blocks-whole-day"
+                      label={t("ReservationUnitEditor.blocksWholeDay")}
+                      onChange={(e) => onChange(e.target.value)}
+                      checked={value != null && value === "blocks-whole-day"}
+                    />
+                    <RadioButton
+                      id="buffer-times-set"
+                      value="buffer-times-set"
+                      label={t("ReservationUnitEditor.setBufferTime")}
+                      onChange={(e) => onChange(e.target.value)}
+                      checked={value != null && value === "buffer-times-set"}
+                    />
+                  </SelectionGroup>
+                )}
+              />
+            </Span12>
+            {watch("reservationBlockWholeDay") === "buffer-times-set" && (
+              <BufferWrapper>
+                <Span6>
```

Comment:
I'd check if you can remove the `Span6` here (they are bad anyway, and create extra divs), if we are defining a custom grid wrapper for it already.

Reply:
Yeah, it was enough to define the custom grid wrapper as `grid-template-columns: repeat(2, 1fr);` - fixed now.